pkg/controller/apischeme/apischeme_controller.go

```diff
@@ -127,12 +112,39 @@ func (r *ReconcileAPIScheme) Reconcile(request reconcile.Request) (reconcile.Res
 		return reconcile.Result{}, nil
 	}
 
-	ownerTags, err := utils.AWSOwnerTag(r.client)
+	// Does the Service exist already?
+	found := &corev1.Service{}
+	err = r.client.Get(context.TODO(), types.NamespacedName{Name: instance.Spec.ManagementAPIServerIngress.DNSName, Namespace: "openshift-kube-apiserver"}, found)
 	if err != nil {
-		reqLogger.Error(err, "Couldn't get the cluster owner tags")
-		SetAPISchemeStatus(instance, "Couldn't reconcile", "Couldn't get the cluster owner tags", cloudingressv1alpha1.ConditionError)
-		r.client.Status().Update(context.TODO(), instance)
-		return reconcile.Result{}, err
+		if errors.IsNotFound(err) {
+			// need to create it
+			dep := r.newServiceFor(instance)
+			err = r.client.Create(context.TODO(), dep)
+			if err != nil {
+				reqLogger.Error(err, "Failure to create new Service")
+				return reconcile.Result{}, err
+			}
+			// Reconcile again to get the new Service and give AWS time to create the ELB
+			reqLogger.Info("Service was just created, so let's try to requeue to set it up")
+			return reconcile.Result{Requeue: true, RequeueAfter: 10 * time.Second}, nil
+		} else if err != nil {
+			reqLogger.Error(err, "Couldn't get the Service")
+			return reconcile.Result{}, err
+		}
+	}
+	// Reconcile the access list in the Service
+	if !sliceEquals(found.Spec.LoadBalancerSourceRanges, instance.Spec.ManagementAPIServerIngress.AllowedCIDRBlocks) {
```

Comment:
Just had a question about this part, coming from how we do things in `managed-velero-operator`.
There, the pattern is usually to create the `dep` object just before `r.client.Get()`, and then at this point do a deep equal between `dep` and `found`.

Was avoiding a full `Service` comparison intentional?

`dep.Spec.LoadBalancerSourceRanges` is the same as `instance.Spec.ManagementAPIServerIngress.AllowedCIDRBlocks`, so you'd get the same CIDR mismatch detection by comparing `dep` and `found`, but also catch any other inconsistencies in `found`.

Reply:
@lisa answered privately:
> it was intentional because the thing we care about is only the allowed cidr blocks, which is a []string. returning a new service object only to compare that seems excessive.